proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,136 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the RET is `float`, and the resource type is `RWBuffer`. The
+resource type is not a `RawBuffer` variant, and so there is a distinct set
+of rules that define valid RETs for this resource type.
+
+RETs for non-`RawBuffer` variants may include basic types (ints and uints of sizes 16
+and 32, as well as half, float, vectors, and matrices of 4 elements or fewer).

```

Comment:
> vectors, and matrices 

does the element type matter for vectors and matrices?

Reply:
I'll add that the element type may not exceed 32 bits.

Reply:
I'm going to mark this as resolved and cover it in a new conversation.